src/Tribe/Process/Queue.php

```diff
@@ -127,12 +128,52 @@ public static function action() {
 	}
 
 	/**
-	 * {@inheritdoc}
+	 * Whether a queue process is stuck or not.
+	 *
+	 * A queue process that has not been doing anything for an amount
+	 * of time is considered "stuck".
+	 *
+	 * @since TBD
+	 *
+	 * @param string $queue_id The queue process unique identifier.
+	 *
+	 * @return bool
 	 */
-	public function delete( $key ) {
+	public static function is_stuck( $queue_id ) {
+		$queue_status = self::get_status_of( $queue_id );
+		$is_stuck     = false;
+
+		/**
+		 * Filters the maximum allowed time a queue process can go without updates
+		 * before being considered stuck.
+		 *
+		 * @since TBD
+		 *
+		 * @param int $time_limit A value in seconds, defaults to 5'.
+		 */
+		$limit = (int) apply_filters( 'tribe_process_queue_time_limit', 300 );
+
+		if ( ! empty( $queue_status['last_update'] ) && is_numeric( $queue_status['last_update'] ) ) {
+			$is_stuck = time() - (int) $queue_status['last_update'] > $limit;
+		} else {
+			$queue_status['last_update'] = time();
+			set_transient( $queue_id . '_meta', $queue_status->to_array(), DAY_IN_SECONDS );
+		}
+
+		return $is_stuck;
```

Comment:
Not actually sure if this would be useful in practice, going to ask anyway: any merit in making the return value itself filterable (and not just $limit)?

Reply:
Filtering results is rarely a bad idea; I can think of a scenario where, to clean all queues (whether they are active or not) one could filter the `stuck` status to clean them all.